### PR TITLE
fix: assign new oauth users to default org

### PR DESCRIPTION
**Pull request by Emyrk**
Begins work on https://github.com/coder/coder/issues/11922

This is not a final solution, as we eventually want to be able to map to different orgs. 
This PR just makes it so multi-org does not break oauth/oidc.

### CODE REVIEW

File: coderd/userauth.go [hunk 1]
```diff
@@ -1343,13 +1343,12 @@ func (api *API) oauthLogin(r *http.Request, params *oauthLoginParams) ([]*http.C
...
-				// support is added, we should enable a configuration map of user
-				// email to organization.
-				organizationID = organizations[0].ID
-			}
+			organization, _ := tx.GetDefaultOrganization(dbauthz.AsSystemRestricted(ctx))
+
+			// Add the user to the default organization.
+			// Once multi-organization we should check some configuration to see
+			// if we should add the user to a different organization.
+			organizationID = organization.ID
```
Comment (Emyrk) [inline coderd/userauth.go:1351]: This is annoyingly allowed to be `uuid.Nil` in unit tests. I am not going to change the behavior here, but we should really fix that so we don't have an edge case living for the sake of tests.

Comment (mafredri) [reply coderd/userauth.go:1351]: Fine not to fix here, but can we add a comment for the ignored err above (`tx.GetDefaultOrganization`) about this? Comment is just a band-aid though, we definitely want to nip this in the bud before too long.

**Review by mafredri — APPROVED** [review]
